pkg/testidentification/ocp_variants_test.go

```diff
@@ -17,78 +17,78 @@ func Test_openshiftVariants_IdentifyVariants(t *testing.T) {
 		{
 			name:    "periodic-ci-openshift-hypershift-main-periodics-conformance-aws-ovn-4-12",
 			release: "4.12",
-			want:    []string{"aws", "amd64", "ovn", "ha", "hypershift"},
+			want:    []string{"amd64", "aws", "ha", "hypershift", "ovn"},
```

Comment:
I think we have testify vendored here, https://pkg.go.dev/github.com/stretchr/testify/assert#ElementsMatch instead of reflection probably works more reliably.  The new version of Sets from apimachinery doesn't guarantee order anymore IIRC

Reply:
Thanks for fixing all this btw

Reply:
Thanks for the review. I am keeping this on hold as we have to get a confirmation about adding `agent` as a new variant. This is because there are some tests that we are not running in our workflow in CI. Once that is resolved, I will address the suggestions on this PR.